UIforETW/Utility.cpp

```diff
@@ -594,69 +871,171 @@ void SetCurrentThreadName(PCSTR threadName)
 #pragma warning(pop)
 
 
-void CopyStartupProfiles(const std::wstring& exeDir, bool force)
+void CopyStartupProfiles(const std::wstring& exeDir, const bool force)
 {
-	const wchar_t* fileName = L"\\Startup.wpaProfile";
 
-	// First copy the WPA 8.1 startup.wpaProfile file
-	wchar_t documents[MAX_PATH];
-	const BOOL getMyDocsResult = SHGetSpecialFolderPathW(NULL, documents, CSIDL_MYDOCUMENTS, TRUE);
-	UIETWASSERT(getMyDocsResult);
-	if (force)
-		outputPrintf(L"\n");
-	if (getMyDocsResult)
+	// WPA 8.1 stores startup.wpaProfile file in Documents/WPA Files
+	copyWPAProfileToDocuments(force);
+
+	copyWPAProfileToLocalAppData(exeDir, force);
+}
+
+_Success_(return)
+bool OpenRegKey( _Out_ HKEY* const key, _In_ const HKEY root, PCWSTR const subkey )
+{
+	const LONG openResult = ::RegOpenKeyExW(root, subkey, 0, KEY_ALL_ACCESS, key);
+	if (openResult == ERROR_SUCCESS)
+		return true;
+	outputPrintf(L"Failed to open registry key `%s`.\n", subkey);
+	debugLastError();
+	return false;
+}
+
+void CloseFindHandle(_In_ _Pre_valid_ _Post_ptr_invalid_ const HANDLE handle, PCWSTR const directory)
+{
+	const BOOL findClose = ::FindClose(handle);
+	if (findClose)
+		return;
+	outputPrintf(L"FindClose (for directory: `%s`) failed.\n", directory);
+	debugLastError();
+	std::terminate();
+}
+
+void CloseValidHandle(_In_ _Pre_valid_ _Post_ptr_invalid_ const HANDLE handle)
+{
+	const BOOL handleClosed = ::CloseHandle(handle);
+	if (handleClosed == 0)
+		std::terminate();//Logic bug!
+}
+
+void CloseRegKey(_In_ _Pre_valid_ _Post_ptr_invalid_ const HKEY key, PCWSTR const keyName)
+{
+	const LONG closeKey = ::RegCloseKey(key);
+	if (closeKey == ERROR_SUCCESS)
+		return;
+	outputPrintf(L"Failed to close registry key `%s`.\n", keyName);
+	debugLastError();
+}
+
+void ClipboardClose()
+{
+	const BOOL closeResult = ::CloseClipboard( );
+	if (closeResult)
+		return;
+	debugPrintf(L"Failed to close the clipboard!\n");
+	debugLastError( );
+}
+
+bool ClipboardOpen()

```

Comment:
This name does not suggest that it opens the clipboard. It sounds like it queries the open status of the clipboard.